src/main/java/edu/harvard/iq/dataverse/AuxiliaryFileServiceBean.java

```diff
@@ -119,6 +134,35 @@ public AuxiliaryFile lookupAuxiliaryFile(DataFile dataFile, String formatTag, St
             return null;
         }
     }
+    
+    @SuppressWarnings("unchecked")
+    public List<AuxiliaryFile> listAuxiliaryFiles(DataFile dataFile, String origin) {
+        
+        Query query = em.createQuery("select object(o) from AuxiliaryFile as o where o.dataFile.id = :dataFileId and o.origin = :origin");
+                
+        query.setParameter("dataFileId", dataFile.getId());
+        query.setParameter("origin", origin);
+        try {
+            List<AuxiliaryFile> retVal = (List<AuxiliaryFile>)query.getResultList();
+            return retVal;
+        } catch(Exception ex) {
+            return null;
```

Comment:
Should we handle the exception from getResultList somehow?

Reply:
Since the exceptions for getResultList(); all indicate real server errors, I've removed the try block so they bubble up to be reported via the API. I also changed the existing getSingleResult() call where one exception (NoResultException) is OK and should return null whereas other exceptions represent a real server error.